modules/jooby-metrics/src/main/java/io/jooby/metrics/MetricsModule.java

```diff
@@ -0,0 +1,277 @@
+    metrics.put(name, metric);
+    return this;
+  }
+
+  /**
+   * Append a metric to the {@link MetricRegistry}. The metric will be resolved by Guice. This call
+   * is identical to {@link MetricRegistry#register(String, Metric)}.
+   *
+   * @param name Name of the metric.
+   * @param metric A metric object.
+   * @param <M> Metric type.
+   * @return This metrics module.
+   */
+  public <M extends Metric> MetricsModule metric(final String name, final Class<M> metric) {
+    metricClasses.put(name, metric);
+    return this;
+  }
+
+  /**
+   * Append a health check to the {@link HealthCheckRegistry}. This call is identical to
+   * {@link HealthCheckRegistry#register(String, HealthCheck)}.
+   *
+   * @param name Name of the check.
+   * @param check A check object.
+   * @return This metrics module.
+   */
+  public MetricsModule healthCheck(final String name, final HealthCheck check) {
+    healthChecks.put(name, check);
+    return this;
+  }
+
+  /**
+   * Append a health check to the {@link HealthCheckRegistry}. The metric will be resolved by Guice.
+   * This call is identical to {@link HealthCheckRegistry#register(String, HealthCheck)}.
+   *
+   * @param name Name of the check.
+   * @param check A check object.
+   * @param <H> {@link HealthCheck} type.
+   * @return This metrics module.
+   */
+  public <H extends HealthCheck> MetricsModule healthCheck(final String name, final Class<H> check) {
+    healthCheckClasses.put(name, check);
+    return this;
+  }
+
+  /**
+   * Append a {@link Reporter} to the {@link MetricRegistry}.
+   *
+   * @param callback Reporter callback.
+   * @return This metrics module.
+   */
+  public MetricsModule reporter(final BiFunction<MetricRegistry, Config, Reporter> callback) {
+    this.reporters.add(requireNonNull(callback, "Callback is required."));
+    return this;
+  }
+
+  /**
+   * Append a {@link Reporter} to the {@link MetricRegistry}.
+   *
+   * @param callback Reporter callback.
+   * @return This metrics module.
+   */
+  public MetricsModule reporter(final Function<MetricRegistry, Reporter> callback) {
+    return reporter((registry, conf) -> callback.apply(registry));
+  }
+
+  @Override
+  public void install(@Nonnull Jooby application) {
+    MetricHandler metricHandler = new MetricHandler();
+    application.route("GET", this.pattern + "/metrics", metricHandler);
+    application.route("GET", this.pattern + "/metrics/:type", metricHandler);
+    application.route("GET", this.pattern + "/healthcheck", new HealthCheckHandler());
+
+    routes.forEach(r -> r.accept(application));
+
+    ServiceRegistry registry = application.getServices();
+
+    registry.putIfAbsent(MetricRegistry.class, metricRegistry);
+    registry.putIfAbsent(HealthCheckRegistry.class, healthCheckRegistry);
+
+    final Set<Reporter> reporters = new HashSet<>();
+
+    application.onStarted(() -> {
+      metrics.forEach((name, instance) -> {
```

Comment:
Instances should not run on application started. We already have them so it is safe to do it on boot/init.